tutorials/tour/_posts/2017-02-13-implicits.md

```diff
@@ -0,0 +1,84 @@
+---
+layout: tutorial
+title: Implicits
+
+disqus: true
+
+tutorial: scala-tour
+categories: tour
+num: 26
+next-page: implicit-conversions
+previous-page: explicitly-typed-self-references
+---
+Implicits allow for automatic application of code when an explicit application isn't supplied.
+
+## Implicit parameters
+
+Implicit parameters allow for the caller to omit an argument if an implicit one is in scope. Use the `implicit` keyword to make a value, object, or expression implicit. You also use it to make the parameter list implicit.
+```tut
+class Greeting(val greeting: String) {
+  def greet(name: String) = s"$greeting, $name"
+}
+implicit val standardGreeting = new Greeting("Hello")
+
+
+def printGreeting(name: String)(implicit greeting: Greeting) = greeting.greet(name)
+
+
+printGreeting("Franchesca")  // Hello, Franchesca
+printGreeting("Fred")(new Greeting("Good day"))  // Good day, Fred
+```
+The `implicit val standardGreeting` is a value that can be supplied as an argument to an implicit parameter automatically. In the method `printGreeting`, the parameter `greeting` is implicit. This means that the caller can either supply an argument normally or skip it. With `greet("Franchesca")`, the compiler doesn't see a greeting but it notices that `greeting` is an implicit parameter so it searches the current scope for an implicit `Greeting` and finds `standardGreeting`.
+
+This becomes useful when you have a lot of similar arguments to function calls throughout your program. However, implicits can make code more difficult to understand because it's not always obvious where they're defined if you import them from another module with a wildcard (e.g. `import MyPredef._`).
```

Comment:
This is absolutely not the main problem I have with implicits.  The _primary_ problem is that they hide an argument that can impact your program, making it less obvious what your methods depend upon because the method call doesn't show you that anything is going on!  The _secondary_ problem is that once you realize a method actually has an implicit parameter, it can sometimes be difficult to track down where it is because it might be from a wildcard import, not something you typed in.

Reply:
As a newbie, just knowing to say `import cats._` is the hard part.